### PR TITLE
Add offset versions of element constraints and enable them for FlatZinc

### PR DESCRIPTION
**Pull request by guidotack**

### CODE REVIEW

File: gecode/int/element.cpp [hunk 1]
```diff
@@ -74,6 +108,27 @@ namespace Gecode {
...
+    if (c.size() == 0)
+      throw TooFewArguments("Int::element");
+    Limits::check(x1,"Int::element");
+    GECODE_POST;
+    for (int i=0; i<c.size(); i++)
+      Limits::check(c[i],"Int::element");
+    ConstIntView cx1(x1);
+    if (offset==0) {
+      GECODE_ES_FAIL(
+                     (Element::post_int<IntView,ConstIntView>(home,c,x0,cx1)));
```
Comment (zayenz) [inline gecode/int/element.cpp]: Non-needed newline.

File: gecode/int/element.cpp [hunk 2]
```diff
@@ -74,6 +108,27 @@ namespace Gecode {
...
+    } else {
+      OffsetView ov(x0,offset);
+      GECODE_ES_FAIL(
```
Comment (zayenz) [inline gecode/int/element.cpp]: Non-needed newline.

File: gecode/int/element.hh [hunk 3]
```diff
@@ -316,6 +316,35 @@ namespace Gecode { namespace Int { namespace Element {
...
   };
 
+ /**
+  * \brief Domain consistent pair-with-offsets propagator
+  *
+  * Requires \code #include <gecode/int/element.hh> \endcode
+  *
+  * \ingroup FuncIntProp
+  */
+ class GECODE_VTABLE_EXPORT PairWithOffsets
```
Comment (zayenz) [inline gecode/int/element.hh:326]: Is there some reason that I'm missing that the code is copied instead of Pair templatized for the view?

**Review by zayenz — APPROVED** [review]
I think it looks good. The extraneous newlines should probably be removed, and I am curious of the copied Pair code.

File: gecode/int/element.hh (hunk 3 above)
Comment (guidotack) [reply gecode/int/element.hh:326]: Pair uses an OffsetView internally, so in a template version we'd need to be able to create an OffsetView of an OffsetView, which seemed more difficult to support than just having two versions of Pair.

Comment (zayenz) [reply gecode/int/element.hh:326]: Ahh, naturally! I missed that when checking.

**Pull request merged**